src/compass/utils/helpers.py

```diff
@@ -338,27 +338,25 @@ def open_raster(filename, band=1):
     raster: np.ndarray
         Numpy array containing the raster band to open
     '''
+    error_channel = journal.error('helpers.open_raster')
+    if not os.path.isfile(filename):
+        err_str = f'{filename} '
+        error_channel.log(err_str)
+        raise FileNotFoundError(err_str)
+
     try:
         ds = gdal.Open(filename, gdal.GA_ReadOnly)
         arr = ds.GetRasterBand(band).ReadAsArray()
         return arr
-    except:
-        pass
-
-    # GDAL reads 1st 2 bytes of ENVI binary to determine file type. If 1st
-    # bytes of flat binary is that of a jpeg but the binary is not then GDAL
-    # throws a libjpeg runtime error. Follow specifically tries to load as an
-    # ENVI file.
-    try:
+    except AttributeError:
```

Comment:
Below is the screenshot I've captured when the error taken place.
I might be wrong, but it looks like the "fake JPEG" situation with `gdal.Open()` throws `RuntimeError`, I'm not sure why `AttributeError` is being caught. Can you explain?

<img width="2120" alt="Screenshot 2023-07-14 at 19 01 09" src="https://github.com/opera-adt/COMPASS/assets/27862199/1df86de4-b437-4278-b9ed-7b03dc140924">

Reply:
@seongsujeong can you provide a path to the `rdr2geo` output for debugging? Thanks

Reply:
The output files are in aurora. I'll send you the path.

Reply:
Failed also on my end to regenerate the raised error I've screenshot'd above. Note that that error was from COMPASS before cal/val release.

Regarding the array in Out[3], maybe you have `gdal.OpenEx` in `open_raster()` to force loading `height.rdr` as ENVI?

One more thing I'm not sure is if `gdal.Open()` actually raises an error when it fails of load rasters. GDAL Raster API tutorial [here](https://gdal.org/tutorials/raster_api_tut.html) does not mention about the error in Python.

I also tried a toy code below, but failed to print out the error:

```python
from osgeo import gdal


path_envi = 'height.rdr'

try:
    ds_envi = gdal.Open(path_envi, gdal.GA_ReadOnly)
    arr = ds_envi.ReadAsArray()

except Exception as e:
    print(f'Error - {e}')
```

Below is output from terminal:
>ERROR 1: libjpeg: Unsupported marker type 0xbf
Error - 'NoneType' object has no attribute 'ReadAsArray'

Maybe `gdal.Open()` just returns `None` when it fails to load the raster?

Reply:
Yes, `gdal` returns `None` a raster fails to load.

I get the same as you when I run your toy code.

Reply:
Looks like `print(f'Error - {e}')` in the toy code above gives a bit of confusion.

```python
from osgeo import gdal


path_envi = 'height.rdr'

try:
    ds_envi = gdal.Open(path_envi, gdal.GA_ReadOnly)
    arr = ds_envi.ReadAsArray()

except Exception as e:
    print(f'Error raised - {e}')

```
>ERROR 1: libjpeg: Unsupported marker type 0xbf
Error raised - 'NoneType' object has no attribute 'ReadAsArray'

In here the error is being raised not because `goal.Open()` is failed but because attempting to load data from `None`. 
That being case I think it makes sense to raise `AttributeError`

```python
from osgeo import gdal


path_envi = 'height.rdr'

try:
    ds_envi = gdal.Open(path_envi, gdal.GA_ReadOnly)
    arr = ds_envi.ReadAsArray()

except AttributeError:
    print('Attribute eror raised')

except Exception as e:
    print(f'Error raised - {e}')

```
>ERROR 1: libjpeg: Unsupported marker type 0xbf
Attribute eror raised

After all, it looks like the code here was already good as it is. Sorry for raising a false alarm. Please consider this comment as resolved.